src/pages/studyView/table/ClinicalEventTypeCountTable.tsx

```diff
@@ -0,0 +1,412 @@
+        width: 300,
+        showAddRemoveAllButtons: false,
+        cancerGeneFilterEnabled: false,
+    };
+
+    @observable protected sortBy: ClinicalEventTypeCountColumnKey;
+
+    getDefaultColumnDefinition = (
+        columnKey: ClinicalEventTypeCountColumnKey,
+        columnWidth: number,
+        cellMargin: number
+    ) => {
+        const defaults: {
+            [key in ClinicalEventTypeCountColumnKey]: Column<
+                ClinicalEventTypeCount
+            >;
+        } = {
+            [ClinicalEventTypeCountColumnKey.CLINICAL_EVENT_TYPE]: {
+                name: columnKey,
+                headerRender: () => (
+                    <TreatmentGenericColumnHeader
+                        margin={cellMargin}
+                        headerName={columnKey}
+                    />
+                ),
+                render: (data: ClinicalEventTypeCount) => (
+                    <ClinicalEventTypeCountColumnCell row={data} />
+                ),
+                sortBy: (data: ClinicalEventTypeCount) => data.eventType,
+                defaultSortDirection: 'asc' as 'asc',
+                filter: filterClinicalEventTypeCountCell,
+                width: columnWidth,
+            },
+            [ClinicalEventTypeCountColumnKey.COUNT]: {
+                tooltip: (
+                    <span>Distinct Number of Patients with Event Type</span>
+                ),
+                name: columnKey,
+                headerRender: () => (
+                    <TreatmentGenericColumnHeader
+                        margin={cellMargin}
+                        headerName={columnKey}
+                    />
+                ),
+                render: (data: ClinicalEventTypeCount) =>
+                    this.createNubmerColumnCell(data, 28),
+                sortBy: (data: ClinicalEventTypeCount) =>
+                    data.count + toNumericValue(data.eventType),
+                defaultSortDirection: 'desc' as 'desc',
+                filter: filterClinicalEventTypeCountCell,
+                width: columnWidth,
+            },
+        };
+        return defaults[columnKey];
+    };
+
+    @computed
+    get columnsWidth() {
+        return _.reduce(
+            this.props.columns,
+            (acc, column) => {
+                acc[column.columnKey] = correctColumnWidth(
+                    (column.columnWidthRatio
+                        ? column.columnWidthRatio
+                        : DEFAULT_COLUMN_WIDTH_RATIO[column.columnKey]) *
+                        this.props.width
+                );
+                return acc;
+            },
+            {} as { [key in ClinicalEventTypeCountColumnKey]: number }
+        );
+    }
+
+    @computed
+    get cellMargin() {
+        return _.reduce(
+            this.props.columns,
+            (acc, column) => {
+                acc[column.columnKey] = 0;
+                return acc;
+            },
+            {} as { [key in ClinicalEventTypeCountColumnKey]: number }
```

Comment:
you can avoid using AS here by putting the type on the acc argument  of the callback.   i.e. 
```
this.props.columns.reduce((acc:TYPE, column)=>...)
```